velox/experimental/cudf/exec/CudfFilterProject.cpp

```diff
@@ -40,13 +41,64 @@ void debugPrintTree(
     debugPrintTree(input, indent + 2);
   }
 }
+
+bool checkAddIdentityProjection(
+    const core::TypedExprPtr& projection,
+    const RowTypePtr& inputType,
+    column_index_t outputChannel,
+    std::vector<exec::IdentityProjection>& identityProjections) {
+  if (auto field = core::TypedExprs::asFieldAccess(projection)) {
+    const auto& inputs = field->inputs();
+    if (inputs.empty() ||
+        (inputs.size() == 1 &&
+         dynamic_cast<const core::InputTypedExpr*>(inputs[0].get()))) {
+      const auto inputChannel = inputType->getChildIdx(field->name());
+      identityProjections.emplace_back(inputChannel, outputChannel);
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// Split stats to attrbitute cardinality reduction to the Filter node.
+std::vector<exec::OperatorStats> splitStats(
```

Comment:
Is this confirmed to be wired in? When we replace `FilterProject` with `CudfFilterProject`, did the stats splitter made for the former work for the latter? It would be nice if this was confirmed to be working, but in case it's not, this can be replaced with a TODO.

Reply:
yes